internal/controller/mantlebackup_controller.go

```diff
@@ -417,12 +417,26 @@ func (r *MantleBackupReconciler) reconcileAsStandalone(ctx context.Context, back
 	return ctrl.Result{}, nil
 }
 
+func (r *MantleBackupReconciler) reconcileAsStandalone(ctx context.Context, backup *mantlev1.MantleBackup) (ctrl.Result, error) {
+	result, err := r.reconcilePre(ctx, backup)
+	if err != nil || !result.IsZero() {
+		return result, err
+	}
+
+	return r.primaryCleanup(ctx, backup)
+}
+
 func (r *MantleBackupReconciler) reconcileAsPrimary(ctx context.Context, backup *mantlev1.MantleBackup) (ctrl.Result, error) {
-	result, err := r.reconcileAsStandalone(ctx, backup)
+	result, err := r.reconcilePre(ctx, backup)
 	if err != nil || !result.IsZero() {
 		return result, err
 	}
-	return r.replicate(ctx, backup)
+
+	if !backup.IsSynced() {
+		return r.replicate(ctx, backup)
```

Comment:
Is it OK not to check?
```suggestion
		result, err := r.replicate(ctx, backup)
		if err != nil || !result.IsZero() {
			return result, err
		}		
```

Reply:
I think this check is unnecessary. Whether `replicate` returns an error or not, we must stop reconciliation here and return the values because we can't perform `secondaryCleanup`.

Reply:
I've changed my mind.